### DISCUSSION TITLE
Is it possible to set string length through `source_options.columns` configuration?

### DISCUSSION THREAD

**Discussion by isaacrlee** [slingdata-io/sling-cli#343] [Q&A] [answered]
Similar to decimal with `decimal(<precision>, <scale>)` is it possible to set length for string types in configuration e.g. `string(16)`?

Reading the source code [here](https://github.com/slingdata-io/sling-cli/blob/main/core/dbio/iop/datatype.go#L804), I see the max length gets parsed but I haven't had luck getting the length to actually be set.

**Comment by flarco** [comment, marked as answer]
Yes, this was the idea at first. Using this [mapping](https://github.com/slingdata-io/sling-cli/blob/main/core/dbio/templates/types_general_to_native.tsv), looking at the `string` type (line 11), the mapped value used to be `varchar()` instead of `varchar(65500)` / `varchar(max)`, where sling would fill the `()` will a number, but it ended up causing issues, so the max number was set. I'll need to take another stab at it.

**Reply by flarco** [reply]
@isaacrlee I pushed some changes for this. Do you mind testing on your local machine? The README has build instructions. Just need go installed.
use branch from https://github.com/slingdata-io/sling-cli/pull/344

**Reply by isaacrlee** [reply]
@flarco  Thanks for implementing this so quickly. I tried out #344 and read through the code.

I'm still not seeing the max length set but that might be because of my specific case; I'm trying to replicate from Snowflake to SQL Server. Is there a reason Snowflake isn't included with the changes (seeing it doesn't have `varchar()` [here](https://github.com/slingdata-io/sling-cli/blob/75a83a6c0a3fc72d3aaf502f0b9a14feba094fae/core/dbio/templates/types_general_to_native.tsv))?

Using `target_options.table_ddl` did work at the cost of having to set every column type, so that's a possible a workaround.

Also, unrelated to this discussion but in my testing of #344 the `table_keys.index` feature for #342 and #226 worked great!

**Reply by flarco** [reply]
Ok, pushed changes, should be good now.

**Reply by isaacrlee** [reply]
@flarco Just tested the changes and it works for the Snowflake source case now, thanks!
